### PR TITLE
Mongodb optimization. Alina Nazarovich

### PR DESCRIPTION
**Pull request by AlinaNH**
https://travis-ci.org/github/AlinaNH/db-assignments/builds/717703909

### CODE REVIEW

File: task/mongo-optimization-task.js [hunk 1]
```diff
@@ -13,6 +13,8 @@ const ObjectId = require('mongodb').ObjectID;
  * */
 async function before(db) {
     await db.collection('opportunities').createIndex({'initiativeId': 1});
+    await db.collection('opportunities').createIndex({"contacts.questions.category_id": 1});
```
Comment (IlyaMokin) [inline task/mongo-optimization-task.js]: in one `$match` only one index can be used. Try to make a compound index with a few fields

File: task/mongo-optimization-task.js [hunk 2]
```diff
@@ -240,22 +226,22 @@ async function task_3_1(db) {
         {
             "$lookup" : {
                 "from" : "clientCriteria",
-                "localField" : "criteria_value",
-                "foreignField" : "value",
+                let: { item: "$criteria_value" },
+                pipeline: [
+                    {
+                        "$match" : {
```
Comment (IlyaMokin) [inline task/mongo-optimization-task.js:231]: do you really need 2 `$match` sections?

**Comment by AlinaNH** [comment]
@IlyaMokin Hello! Thank you for your comments. Done - https://travis-ci.org/github/AlinaNH/db-assignments/builds/725031661

File: task/mongo-optimization-task.js [hunk 3]
```diff
@@ -240,22 +225,18 @@ async function task_3_1(db) {
...
+                            "versions.initiativeId" : ObjectId("58af4da0b310d92314627290"),
+                            $expr: { $eq: [ "$value",  "$$item" ] }
+                        }
+                    }
```
Comment (IlyaMokin) [inline task/mongo-optimization-task.js:238]: Additional projection, compound index and `limit` at the end can make the query faster.

Comment (AlinaNH) [reply task/mongo-optimization-task.js:238]: @IlyaMokin thank you! done - https://travis-ci.org/github/AlinaNH/db-assignments/builds/726557102

**Comment by vramaniuk** [comment]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**